web_generator/lib/src/interop_gen/transform/transformer.dart

```diff
@@ -72,6 +79,537 @@ class Transformer {
     nodes.add(node);
   }
 
+  InterfaceDeclaration _transformInterface(TSInterfaceDeclaration interface) {
```

Comment:
This is also very similar to the code below in `_transformClass`, can we share the code?

Reply:
To elaborate, the implementation of this method and `_transformClass` do very similar things and have a very similar structure, so it may help to have a helper method here.